### ISSUE TITLE
Use subprocess to adapt in serial

### ISSUE THREAD

**Issue by jwallwork23** [mesh-adaptation/goalie#96]
ParMmg doesn't support parallel adaptation in 2D. However, solving 2Ds PDEs in parallel is a standard activity of Firedrake. We should use Python `subprocess` to enable serial adaptation during a parallel run.

**Comment by jwallwork23** [comment]
https://docs.python.org/3/library/subprocess.html

**Comment by ddundo** [comment]
Small comment here with my experience, maybe irrelevant. I would adapt in serial by doing something like this:

```python
if COMM_WORLD.rank == 0:
      arguments = f'--input temp_chk.h5 --output {output_fpath} --iteration {iteration} --interval {i}'.split()
      subprocess.run(["mpirun", "-n", "1", "python", "adapt_script.py"] + arguments)
COMM_WORLD.barrier()
```

So I would have a separate script for adaptation that I would run in serial. If I did not do this, and I adapted under the `if COMM_WORLD.rank == 0` statement (and not in a separate script that I would call using `subprocess`), it would be extremely slow. 

You recommended that I check the flame graph, but I gave up on using parallel so I never did it - but I can if we encounter this again / you think it's useful.
